splitio/storage/redis.py

```diff
@@ -16,101 +16,151 @@
             _LOGGER.debug(raw)
             return splits.from_raw(json.loads(raw)) if raw is not None else None
         except RedisAdapterException:
-            _LOGGER.error('Error fetching split from storage')
+            _LOGGER.error('Error fetching feature flag from storage')
+            _LOGGER.debug('Error: ', exc_info=True)
+            return None
+
+    def get_feature_flags_by_sets(self, flag_sets):
+        """
+        Retrieve feature flags by flag set.
+
+        :param flag_set: Names of the flag set to fetch.
+        :type flag_set: str
+
+        :return: Feature flag names that are tagged with the flag set
+        :rtype: listt(str)
+        """
+        try:
+            sets_to_fetch = []
+            for flag_set in flag_sets:
+                if flag_set not in self._flag_sets and len(self._flag_sets) > 0:
+                    _LOGGER.warning("Flag set %s is not part of the configured flag set list, ignoring the request." % (flag_set))
+                    continue
+                sets_to_fetch.append(flag_set)
+
+            if sets_to_fetch == []:
+                return []
+
+            keys = [self._get_set_key(feature_flag_name) for feature_flag_name in sets_to_fetch]
+            pipe = self._pipe()
+            [pipe.smembers(key) for key in keys]
+            result_sets = pipe.execute()
+            _LOGGER.debug("Fetchting Feature flags by set [%s] from redis" % (keys))
+            _LOGGER.debug(result_sets)
+            to_return = set()
+            [to_return.update(result_set) for result_set in result_sets]
+            return list(to_return)
+        except RedisAdapterException:
+            _LOGGER.error('Error fetching feature flag from storage')
             _LOGGER.debug('Error: ', exc_info=True)
             return None
 
-    def fetch_many(self, split_names):
+    def fetch_many(self, feature_flag_names):
         """
-        Retrieve splits.
+        Retrieve feature flags.
 
-        :param split_names: Names of the features to fetch.
-        :type split_name: list(str)
+        :param feature_flag_names: Names of the features to fetch.
+        :type feature_flag_name: list(str)
 
         :return: A dict with split objects parsed from redis.
         :rtype: dict(split_name, splitio.models.splits.Split)
         """
         to_return = dict()
         try:
-            keys = [self._get_key(split_name) for split_name in split_names]
-            raw_splits = self._redis.mget(keys)
-            _LOGGER.debug("Fetchting Splits [%s] from redis" % split_names)
-            _LOGGER.debug(raw_splits)
-            for i in range(len(split_names)):
-                split = None
+            keys = [self._get_key(feature_flag_name) for feature_flag_name in feature_flag_names]
+            raw_feature_flags = self._redis.mget(keys)
+            _LOGGER.debug("Fetchting feature flags [%s] from redis" % feature_flag_names)
+            _LOGGER.debug(raw_feature_flags)
```

Comment:
do we need this now?

Reply:
Yeah, this is to log all redis calls and response content similar to http log